app/code/Magento/ConfigurableProductGraphQl/Model/Variant/Collection.php

```diff
@@ -143,13 +161,19 @@ private function fetch() : array
             return $this->childrenMap;
         }
 
+        $showOutOfStock = $this->stockConfig->isShowOutOfStock();
+
         foreach ($this->parentProducts as $product) {
             $attributeData = $this->getAttributesCodes($product);
             /** @var ChildCollection $childCollection */
             $childCollection = $this->childCollectionFactory->create();
             $childCollection->setProductFilter($product);
             $childCollection->addAttributeToSelect($attributeData);
 
+            if (!$showOutOfStock) {
```

Comment:
@minesh93, please cover this functionality with API-functional tests
thanks

Examples:
https://github.com/magento/graphql-ce/tree/2.3-develop/dev/tests/api-functional/testsuite/Magento/GraphQl/Customer
https://github.com/magento/graphql-ce/tree/2.3-develop/dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Customer

Guide:
https://devdocs.magento.com/guides/v2.3/graphql/functional-testing.html